atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,787 @@
+            sec_atomsgroup.label = particles_group.pop('label', None)
+            sec_atomsgroup.composition_formula = particles_group.pop('formula', None)
+            particles_subgroup = particles_group.pop('particles_group', None)
+            # set the remaining attributes
+            for particles_group_key in particles_group.keys():
+                unit = None
+                val = particles_group.get(particles_group_key)
+                has_units = hasattr(val, 'units') if val is not None else None
+                if has_units:
+                    unit = val.units
+                    val = val.magnitude
+                sec_atomsgroup.x_h5md_parameters.append(ParamEntry(kind=particles_group_key, value=val, unit=unit))
+            # get the next atomsgroup
+            if particles_subgroup:
+                self.get_atomsgroup_fromh5md(sec_atomsgroup, particles_subgroup)
+
+    def check_metainfo_for_key_and_Enum(self, metainfo_class, key, val):
+        if key in metainfo_class.__dict__.keys():
+            quant = metainfo_class.__dict__.get(key)
+            if quant.get('type') is not None:
+                if type(quant.type).__name__ == 'MEnum':
+                    if val in quant.type._list:
+                        return key
+                    else:
+                        return 'x_h5md_' + key
+                else:
+                    return key
+            else:
+                self.logger.warning(key + 'in ' + metainfo_class + ' is not a Quantity or does not have an associated type.')  # Not sure if this can ever happen
+                return key
+        else:
+            return 'x_h5md_' + key
+
+    @property
+    def parameter_info(self):
+        if self._parameter_info is None:
+            parameters_group = self.h5md_groups.get('parameters')
+            if parameters_group is None:
+                return
+            self._parameter_info = {
+                'force_calculations': {},
+                'workflow': {}
+            }
+
+            def get_parameters_recursive(parameter_group):
+                param_dict = {}
+                for key, val in parameter_group.items():
+                    if type(val) == h5py.Group:
+                        param_dict[key] = get_parameters_recursive(val)
+                    else:
+                        param_dict[key] = self._data_parser.hdf5_getter(parameter_group, key)
+                        # if type(param_dict[key]) == str:
+                        if isinstance(param_dict[key], str):
+                            if key == 'thermodynamic_ensemble':
+                                param_dict[key] = param_dict[key].upper()  # TODO change enums to lower case and adjust Gromacs and Lammps code accordingly
+                            else:
+                                param_dict[key] = param_dict[key].lower()
+                        elif 'int' in type(param_dict[key]).__name__:
```

Comment:
isinstance(value, (int, np.int, np.int32))?

Reply:
indeed, thank you. well I added np.int64 instead of np.int. done.